playwright/_impl/_network.py

```diff
@@ -201,24 +207,56 @@ async def fulfill(
         body: Union[str, bytes] = None,
         path: Union[str, Path] = None,
         contentType: str = None,
-        response: "APIResponse" = None,
+        response: Union["APIResponse", HARResponse] = None,
+        har_fallback: Union[RouteHARFallback, None] = None,
+        har_path: Union[pathlib.Path, str, None] = None,
     ) -> None:
         params = locals_to_params(locals())
         if response:
-            del params["response"]
-            params["status"] = (
-                params["status"] if params.get("status") else response.status
-            )
-            params["headers"] = (
-                params["headers"] if params.get("headers") else response.headers
-            )
             from playwright._impl._fetch import APIResponse
 
-            if body is None and path is None and isinstance(response, APIResponse):
-                if response._request._connection is self._connection:
-                    params["fetchResponseUid"] = response._fetch_uid
-                else:
-                    body = await response.body()
+            if isinstance(response, APIResponse):
```

Comment:
This isn't quite correct. This is checking for `playwright._impl._fetch.APIResponse` but the type that ends up here is `playwright.async_api._generated.APIResponse` (or `playwright.sync_api._generated.APIResponse`) based on debugging `test_should_fulfill_with_any_response`.

We can't import the generated types (and the generated types don't subclass `playwright._impl._fetch.APIResponse`), so I could switch this code to be:

```python
if isinstance(response, dict):
   # fulfilling from HAR response
else:
   # fulfilling from some kind of APIResponse
```

@mxschmitt do you have a recommendation, or am I missing something?